tests/unit/CollectionSpecificOperationTest.php

```diff
@@ -0,0 +1,478 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace tests\loophp\collection;
+
+use ArrayIterator;
+use Closure;
+use Exception;
+use Generator;
+use InvalidArgumentException;
+use loophp\collection\Collection;
+use loophp\collection\Contract\Collection as CollectionInterface;
+use loophp\collection\Operation\Coalesce;
+use loophp\collection\Operation\Current;
+use loophp\collection\Operation\Limit;
+use OutOfBoundsException;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+use tests\loophp\collection\Traits\AssertIdenticalIterable;
+use tests\loophp\collection\Traits\GenericCollectionProviders;
+use function gettype;
+
+/**
+ * @internal
+ * @coversDefaultClass \loophp\collection
+ */
+final class CollectionSpecificOperationTest extends TestCase
```

Comment:
Is this one going to have tests with each of the operations or is it more for specific ones that we feel need specific tests?

Reply:
Indeed, only the more specific one. I'm very bad at naming, do you have a better suggestion ?

Reply:
The name is makes sense I think, so it's all good